packages/sdk-js/src/Identity.ts

```diff
@@ -0,0 +1,456 @@
+    this.didSigners.push(...signers)
+    return this
+  }
+
+  public async addKeypair(
+    ...keypairs: Array<Keypair | KeyringPair>
+  ): Promise<IdentityClass> {
+    const didKeys = this.didDocument.verificationMethod?.map(
+      ({ publicKeyMultibase, id }) => ({
+        ...multibaseKeyToDidKey(publicKeyMultibase),
+        id,
+      })
+    )
+    if (didKeys && didKeys.length !== 0) {
+      await Promise.all(
+        keypairs.map(async (keypair) => {
+          const thisType = 'type' in keypair ? keypair.type : undefined
+          const matchingKey = didKeys?.find(({ publicKey, keyType }) => {
+            if (thisType && thisType !== keyType) {
+              return false
+            }
+            return u8aEq(publicKey, keypair.publicKey)
+          })
+          if (matchingKey) {
+            const id = matchingKey.id.startsWith('#')
+              ? this.did + matchingKey.id
+              : matchingKey.id
+            this.addSigner(
+              ...(await Signers.getSignersForKeypair({
+                keypair,
+                id,
+                type: matchingKey.keyType,
+              }))
+            )
+          }
+        })
+      )
+    }
+    return this
+  }
+
+  public getSigners({
+    verificationMethod,
+    verificationRelationship,
+    algorithm,
+  }: {
+    verificationMethod?: DidUrl | UriFragment
+    verificationRelationship?: string
+    algorithm?: string
+  } = {}): SignerInterface[] {
+    const selectors = [
+      Signers.select.byDid(this.didDocument, { verificationRelationship }),
+    ]
+    if (algorithm) {
+      selectors.push(Signers.select.byAlgorithm([algorithm]))
+    }
+    if (verificationMethod) {
+      selectors.push(Signers.select.bySignerId([verificationMethod]))
+    }
+    return Signers.selectSigners(this.didSigners, ...selectors)
+  }
+
+  public getSigner(
+    criteria: {
+      verificationMethod?: DidUrl | UriFragment
+      verificationRelationship?: string
+      algorithm?: string
+    } = {}
+  ): SignerInterface {
+    const [signer] = this.getSigners(criteria)
+    if (typeof signer === 'undefined') {
+      throw new SDKErrors.NoSuitableSignerError(undefined, {
+        signerRequirements: criteria,
+        availableSigners: this.didSigners,
+      })
+    }
+    return signer
+  }
+}
+
+export type TransactionStrategy<T extends IdentityClass> = (
+  identity: IdentityClass & Identity
+) => Promise<T>
+
+export async function makeIdentity(args: {
+  did?: Did
+  didDocument?: DidDocument
+  signers?: SignerInterface[]
+  keypairs?: Array<Keypair | KeyringPair>
+  resolver?: typeof resolve
+}): Promise<IdentityClass>
+export async function makeIdentity<T extends IdentityClass>(args: {
+  did?: Did
+  didDocument?: DidDocument
+  signers?: SignerInterface[]
+  keypairs?: Array<Keypair | KeyringPair>
+  resolver?: typeof resolve
+  transactionStrategy: TransactionStrategy<T>
+}): Promise<T>
+export async function makeIdentity<T extends IdentityClass>({
```

Comment:
there's `makeIdentity` and `newIdentity`; names could be improved to distinguish them better.